platform/linux-generic/odp_ipsec.c

```diff
@@ -1307,8 +1308,10 @@ static ipsec_sa_t *ipsec_out_single(odp_packet_t pkt,
 	ipsec_sa = _odp_ipsec_sa_use(sa);
 	ODP_ASSERT(NULL != ipsec_sa);
 
-	if ((opt && opt->frag_mode == ODP_IPSEC_FRAG_CHECK) ||
-	    (!opt && ipsec_sa->out.frag_mode == ODP_IPSEC_FRAG_CHECK))
+	frag_mode = ipsec_sa->out.frag_mode;
+	if (opt->flag.frag_mode)
+		frag_mode = opt->frag_mode;
```

Comment:
`frag_mode = opt->flag.frag_mode ? opt->frag_mode : ipsec_sa->out.frag_mode;` might be simpler here.

Reply:
ack